File: src/dataset/cpp/resistor-color-duo_exercism-toemm/openAI/ChatGPT4_resistor_color_duo_v4.cpp
#include "resistor_color_duo.h"

#include <array>
#include <unordered_map>

namespace resistor_color_duo {

constexpr std::array<const char*, 10> encoding{
    "black", "brown", "red", "orange", "yellow", 
    "green", "blue", "violet", "grey", "white"};

constexpr std::unordered_map<std::string_view, int> encoding_map = [] {
    std::unordered_map<std::string_view, int> map;
    for (std::size_t i = 0; i < encoding.size(); ++i) {
        map[encoding[i]] = static_cast<int>(i);
    }
    return map;
}();

constexpr int color_code(std::string_view color) noexcept {
    auto it = encoding_map.find(color);
    return it != encoding_map.end() ? it->second : 0;
}

int value(const std::vector<std::string>& input) noexcept {
    return color_code(input[0]) * 10 + color_code(input[1]);
}

}  // namespace resistor_color_duo

File: src/dataset/typescript/resistor-color-duo_exercism-wackerow/openAI/ChatGPT4_resistor-color-duo_v4.ts
export class ResistorColor {
  private static readonly colorConversion: Record<string, number> = {
    black: 0,
    brown: 1,
    red: 2,
    orange: 3,
    yellow: 4,
    green: 5,
    blue: 6,
    violet: 7,
    grey: 8,
    white: 9,
  };

  constructor(private readonly colors: [string, string]) {}

  value = (): number =>
    ResistorColor.colorConversion[this.colors[0]] * 10 +
    ResistorColor.colorConversion[this.colors[1]];
}